components/ruby/lib/chef-licensing/restful_client/base.rb

```diff
@@ -76,8 +83,24 @@ def list_licenses(params = {})
 
       # a common method to handle the get API calls
       def invoke_get_api(endpoint, params = {})
-        response = invoke_api(ChefLicensing::Config.license_server_url.split(","), endpoint, :get, nil, params)
-        response.body
+        if self.class::CACHE_ENDPOINTS.include?(endpoint) && ChefLicensing::Config.cache_enabled?
+          cache_key = construct_cache_key(endpoint, params)
+          logger.debug "Fetching data from cache for #{cache_key}"
+          @cache_manager.fetch(cache_key) do
+            logger.debug "Cache not found for #{cache_key}"
+            logger.debug "Fetching data from server for #{cache_key}"
+            response = invoke_api(ChefLicensing::Config.license_server_url.split(","), endpoint, :get, nil, params)
```

Comment:
My understanding is that we query the list of license server URLs only once and start using the one that is working for the rest of the process.. has that idealogy changed?

Reply:
so I don't understand why we need to split it everytime.. it is just redundant code ain't it ?

Reply:
We either split it everytime (which would give us only one URL after the first run) or we do a conditional check to see if we did atleast one run and the URL just contains one url. I find the splitting the URL everytime to be more cleaner.